docs-src/content/getting-started/streaming.md

```diff
@@ -101,12 +107,213 @@ audio = open('test.wav', 'rb')
+    stream *portaudio.Stream
+}
+
+// NewMicrophone instantiates a Microphone object with the desired
+// sampling rate and buffer size. When streaming to cubic, the sample
+// rate should be set to the sample rate of the model used. (See below)
+func NewMicrophone(sampleRate, bufferSize uint32) (*Microphone, error) {
+
+    // bufferSize is measured in number of bytes. Since we are capturing
+    // 16 bit audio, each sample is 2 bytes. The microphone has a int16
+    // buffer, so we use the number of samples as its size.
+    numSamples := bufferSize/2
+    mic := Microphone{buffer: make([]int16, numSamples)}
+
+    portaudio.Initialize()
+    stream, err := portaudio.OpenDefaultStream(1, 0, float64(sampleRate), int(numSamples), mic.buffer)
+    if err != nil {
+        return nil, err
+    }
+    mic.stream = stream
+
+    err = mic.stream.Start()
+    if err != nil {
+        return nil, err
+    }
+
+    return &mic, nil
+}
+
+// Read copies N bytes into the passed buffer from the microphone audio buffer
+// where N is the buffer size passed to `cubic.NewClient`. Also, to be compatible
+// with the cubic client, it returns two things : an int representing the number of
+// bytes copied and an error.
+func (mic *Microphone) Read(buffer []byte) (int, error) {
+    err := mic.stream.Read()
+    if err != nil {
+        return 0, err
+    }
+    byteBuffer := new(bytes.Buffer)
+    err = binary.Write(byteBuffer, binary.LittleEndian, mic.buffer)
+    if err != nil {
+        return 0, err
+    }
+    copy(buffer, byteBuffer.Bytes())
+    return len(buffer), nil
+}
+
+// Close shuts the microphone stream down and cleans up
+func (mic *Microphone) Close() {
+    mic.stream.Stop()
+    mic.stream.Close()
+    portaudio.Terminate()
+}
+
+func main() {
+
+    bufferSize := uint32(8192)  // 8192 bytes
+    client, err := cubic.NewClient(serverAddr, cubic.WithInsecure(), cubic.WithStreamingBufferSize(bufferSize))
+    if err != nil {
+        log.Fatal(err)
+    }
+    defer client.Close()
+
+    modelResp, err := client.ListModels(context.Background())
+    if err != nil {
+        log.Fatal(err)
+    }
+
+    // Use the first available model
+    model := modelResp.Models[0]
+
+    // Create microphone stream
+    mic, err := NewMicrophone(model.Attributes.SampleRate, bufferSize)
+    if err != nil {
+        log.Fatal(err)
+    }
+    defer mic.Close()
+
+    cfg := &cubicpb.RecognitionConfig{
+        ModelId: model.Id,
+    }
+
+    // Define a callback function to handle results
+    resultHandler := func(resp *cubicpb.RecognitionResponse) {
+        for _, r := range resp.Results {
+            if r.IsPartial {
+                fmt.Print("\r", r.Alternatives[0].Transcript) // print on same line
+            } else {
+                fmt.Println("\r", r.Alternatives[0].Transcript) // print and move to new line
+            }
+        }
+    }
+
+    err = client.StreamingRecognize(context.Background(), cfg, mic, resultHandler)
```

Comment:
move the microphone creation here, and add a comment that since mic implements io.ReadCloser we can use it directly in the StreamingRecognize function.